myriad-scheduler/src/main/java/org/apache/myriad/scheduler/yarn/interceptor/MyriadCleanupInterceptor.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.myriad.scheduler.yarn.interceptor;
+
+import java.io.IOException;
+
+import org.apache.mesos.Protos.Status;
+import org.apache.myriad.Main;
+import org.apache.myriad.scheduler.MyriadDriverManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Responsible for shutdown Myriad by invoking stopDriver upon the 
+ * Myriad driver {@link org.apache.myriad.Main}
+ */
+public class MyriadCleanupInterceptor extends BaseInterceptor {
```

Comment:
Do we really need cleanup interceptor or it can be part of MyriadInitializationInterceptor - cleanup() method?

Reply:
Hi @yufeldman, it seems that @taojieterry does not respond. In my opinion you do not need a new interceptor, you can take advantage of the existing one. Perhaps with a name change like MyriadHelperInterceptor or MyriadStartStopInterceptor so that the behavior of the class is clear?